src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,900 @@
+        public void SetLed(byte led, byte red, byte green = 0, byte blue = 0)
+        {
+
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetLed, led, red, green, blue };
+            var reply = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Get the 5v circuit voltage
+        /// </summary>
+        /// <returns>5v circuit voltage</returns>
+        public double GetVoltage5V()
+        {
+            var value = SpiRead16(SpiMessageType.GetVoltage5V);
+            return (value / 1000.0);
+        }
+
+        /// <summary>
+        /// Get the battery voltage
+        /// </summary>
+        /// <returns>battery/Vcc voltage</returns>
+        public double GetVoltageBatteryVcc()
+        {
+            var value = SpiRead16(SpiMessageType.GetVoltageVcc);
+            return (value / 1000.0);
+        }
+
+        #endregion
+
+        #region Servo and motors
+
+        /// <summary>
+        /// Move the servo motor to a specific pulse
+        /// </summary>
+        /// <param name="servo">Tjhe servo port Servo1 or Servo2</param>
+        /// <param name="pulseMicroseconds">The pulse in microseconds</param>
+        public void SetServo(ServoPort servo, int pulseMicroseconds)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetServo, (byte)servo, (byte)((pulseMicroseconds >> 8) & 0xFF), (byte)(pulseMicroseconds & 0xFF) };
+            var reply = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor power in percent
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="powerPercent">The power from - 100 to 100, or -128 for float</param>
+        public void SetMotorPower(MotorPort port, int powerPercent)
+        {
+            powerPercent = powerPercent > 127 ? (byte)127 : powerPercent;
+            powerPercent = powerPercent < -128 ? -128 : powerPercent;
+            byte bPower = (byte)(powerPercent & 0xFF);
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetMotorPower, (byte)port, bPower };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor target position in degrees
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="positionDegree">The target position in degree</param>
+        public void SetMotorPosition(MotorPort port, int positionDegree)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetMotorPosition, (byte)port, (byte)((positionDegree >> 24) & 0xFF), (byte)((positionDegree >> 16) & 0xFF), (byte)((positionDegree >> 8) & 0xFF), (byte)(positionDegree & 0xFF) };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor target position KP constant
```

Comment:
what does KP mean?

Reply:
yes. See https://en.wikipedia.org/wiki/PID_controller#PI_controller. I can add in the code the link in this wikipedia page